crates/communication/src/server/mod.rs

```diff
@@ -1,8 +1,72 @@
 mod acceptor;
 mod connection;
 mod outputs;
+mod parameters;
 mod receiver;
 mod runtime;
 mod sender;
 
+use std::{
+    collections::BTreeMap,
+    hash::{Hash, Hasher},
+};
+
 pub use runtime::Runtime;
+use serialize_hierarchy::HierarchyType;
+use tokio::sync::mpsc::Sender;
+
+use crate::messages::Response;
+
+#[derive(Clone, Debug)]
+pub(crate) struct Client {
+    pub id: usize,
+    pub response_sender: Sender<Response>,
+}
```

Comment:
I'm not sure where I'd put this type but it seems to be out of place here...

Reply:
The reasoning for this place is that `Client` is used in `outputs` and `parameters` and therefore needs to be shared in the parent module.

Reply:
Sure, but I think it's complex enough to warrant giving it its own file.

Reply:
Done